src/charm.py

```diff
@@ -67,7 +77,16 @@ def __init__(self, *args):
             strip_prefix=True,
         )
 
+        self.endpoints_provider = HydraEndpointsProvider(self)
+
         self.framework.observe(self.on.hydra_pebble_ready, self._on_hydra_pebble_ready)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+        self.framework.observe(
+            self.on[RELATION_NAME].relation_created, self._on_hydra_relation_changed
```

Comment:
```suggestion
            self.on.endpoint_info_relation_created, self._on_hydra_relation_changed
```

Reply:
I think it's useful to be specific in the relation name here aaand the variable `RELATION_NAME` is not great